### ISSUE TITLE
KDF IKEv1 SPEC: Change algSpecs to capabilities

### ISSUE THREAD

**Issue by smuellerDD** [usnistgov/ACVP#384]
When registering the following IKEv1 definition, the server responds an error:

```
        {
          "algorithm":"kdf-components",
          "mode":"ikev1",
          "prereqVals":[
            {
              "algorithm":"HMAC",
              "valValue":"same"
            },
            {
              "algorithm":"SHA",
              "valValue":"same"
            }
          ],
          "algSpecs":[
            {
              "authenticationMethod":"psk",
              "initiatorNonceLength":[
                64,
                65,
                128,
                177,
                256,
                512
              ],
              "responderNonceLength":[
                64,
                89,
                377,
                777,
                1024
              ],
              "diffieHellmanSharedSecretLength":[
                224,
                384,
                777,
                8192
              ],
              "preSharedKeyLength":[
                8,
                384,
                777,
                8192
              ],
              "hashAlg":[
                "SHA-1",
                "SHA2-256",
                "SHA2-384",
                "SHA2-512"
              ]
            }
          ]
        }
```

Reported server error:

```
{
  "acv_version" : "0.4",
  "error" : "Incorrectly formatted JSON (1:1916): expected field name was not provided: capabilities"
}
```

The spec does not contain any reference to "capabilities".

**Comment by smuellerDD** [comment]
Am Montag, 4. Juni 2018, 18:05:28 CEST schrieb Chris Celi:

Hi Chris,

> This should work if you rename the `algSpecs` property to `capabilities` and
> is a mismatch in our server code.

That works. I leave that issue open to track the spec update.

Ciao
Stephan

**Comment by celic** [comment]
Alright. I'm going to edit the title a bit to more accurately reflect the work that needs to be done.

**Comment by smuellerDD** [comment]
Fixed with pull request #417.
